code/controllers/subsystems/lighting.dm

```diff
@@ -100,59 +110,88 @@ var/datum/controller/subsystem/lighting/SSlighting
 	var/list/curr_corners = corner_queue
 	var/list/curr_overlays = overlay_queue
 
-	while (curr_lights.len)
-		var/datum/light_source/L = curr_lights[curr_lights.len]
-		curr_lights.len--
+	while (curr_lights.len && lq_idex <= curr_lights.len)
```

Comment:
If len == 0, then lq_idex will always be > len. At which point, the first condition is redundant?